[Datasets] Remove "ML Preprocessing" user guide (#32965)
ray-project/ray: doc/source/data/transforming-datasets.rst

>   When using the actor compute strategy, per-row and per-batch UDFs can also be

Reviewer: ```suggestion
  With the actor compute strategy, you can use per-row and per-batch UDFs
```

>   *callable classes*, i.e. classes that implement the ``__call__`` magic method. The

Reviewer: ```suggestion
  *callable classes*, i.e., classes that implement the ``__call__`` magic method. The
```

>   constructor of the class can be used for stateful setup, and will be only invoked once

Reviewer: ```suggestion
  can use the constructor of the class for stateful setup, and it is only invoked once
```

  per worker actor.

>   Callable classes are useful if there is expensive state (such as a neural network) that need to be loaded for the UDF. By using an actor class, the state only needs to be loaded once in the beginning, rather than for each batch.

Reviewer: ```suggestion
  Callable classes are useful if you need to load an expensive state (such as a neural network) for the UDF. By using an actor class, the you only need to load the state once in the beginning, rather than for each batch.
```

    :start-after: __writing_simple_out_row_udfs_begin__
    :end-before: __writing_simple_out_row_udfs_end__

> .. _datasets-groupbys:

Reviewer 2: Shall we put this section to the end of this file?

`Writing User-defined Functions (UDFs)` -> `Configuring Batch Size` -> `Compute Strategy` is a coherent story for writing `map_batches`.

> 

Reviewer 2: what about https://docs.ray.io/en/master/data/dataset-ml-preprocessing.html#random-shuffle and https://docs.ray.io/en/master/data/dataset-ml-preprocessing.html#random-block-order? Should we add here as well?

Author: For `randomize_block_order`, I've added a callout in "Shuffling Data" section of the AIR user guide.

For `random_shuffle`, I'll follow Clark's suggestion:
> As for the shuffling docs, why don't we add a shuffle section to the "Transforming Datasets" feature guide that only details the ds.random_shuffle() API (basically the section that we have in the current "ML Preprocessing" guide), and have it cross-link to the AIR docs on shuffling + call out that local shuffling is a good alternative, but delegate details to the AIR level docs?
